src/main/java/org/project/springweb/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package org.project.springweb.service.shoppingcart;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.project.springweb.dto.cartitem.CreateCartItemRequestDto;
+import org.project.springweb.dto.cartitem.UpdateCartItemRequestDto;
+import org.project.springweb.dto.shoppingcart.ShoppingCartResponseDto;
+import org.project.springweb.exception.DataProcessingException;
+import org.project.springweb.exception.EntityNotFoundException;
+import org.project.springweb.mapper.CartItemMapper;
+import org.project.springweb.mapper.ShoppingCartMapper;
+import org.project.springweb.model.Book;
+import org.project.springweb.model.CartItem;
+import org.project.springweb.model.ShoppingCart;
+import org.project.springweb.model.User;
+import org.project.springweb.repository.book.BookRepository;
+import org.project.springweb.repository.cartitem.CartItemRepository;
+import org.project.springweb.repository.shoppingcart.ShoppingCartRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartResponseDto getShoppingCartForUser(Long userId) {
+        return shoppingCartMapper.toDto(findShoppingCartByUserId(userId));
+    }
+
+    @Override
+    public ShoppingCartResponseDto addCartItem(CreateCartItemRequestDto requestDto, Long userId) {
+        Book book = bookRepository.findById(requestDto.getBookId()).orElseThrow(
+                () -> new EntityNotFoundException("Cannot find book with id "
+                        + requestDto.getBookId())
+        );
+        ShoppingCart shoppingCart = findShoppingCartByUserId(userId);
+        Optional<CartItem> existingCartItem = shoppingCart.getCartItems().stream()
+                .filter(item -> item.getBook().getId()
+                        .equals(book.getId()))
+                .findFirst();
+        CartItem cartItem;
+        if (existingCartItem.isPresent()) {
+            cartItem = existingCartItem.get();
+            cartItem.setQuantity(cartItem.getQuantity() + requestDto.getQuantity());
+            cartItemRepository.save(cartItem);
+        } else {
+            cartItem = cartItemMapper.toEntity(requestDto);
+            cartItem.setShoppingCart(shoppingCart);
+            shoppingCart.getCartItems().add(cartItem);
+        }
```

Comment:
Seems like it's the same logic as for update, so I suggest to call "updateCartItem()" endpoint if user wants to change quantity of item. 
Deal?)